packages/musap-rn-kms/src/agent/MusapKeyManagerSystem.ts

```diff
@@ -0,0 +1,127 @@
+import { IKey, ManagedKeyInfo, MinimalImportableKey, TKeyType } from '@veramo/core';
+import {
+  KeyAlgorithmType,
+  KeyGenReq,
+  MusapKey,
+  MusapModuleType, signatureAlgorithmFromKeyAlgorithm,
+  SignatureAlgorithmType,
+  SignatureFormat,
+  SignatureReq,
+} from '@sphereon/musap-react-native'
+import { SscdType } from '@sphereon/musap-react-native/src/types/musap-types';
+import { KeyManagementSystem } from '@veramo/kms-local';
+import { AbstractPrivateKeyStore } from '@veramo/key-manager';
+import 'react-native-get-random-values'
+import { v4 as uuid } from 'uuid';
+import { TextDecoder } from 'text-encoding';
+import Debug from 'debug'
+
+const debug = Debug('sphereon:musap-rn-kms')
+
+export class MusapKeyManagementSystem extends KeyManagementSystem {
+  private musapKeyStore: MusapModuleType;
+  private sscdType: SscdType;
+
+  constructor(keyStore: MusapModuleType, sscdType?: SscdType) {
+    super(keyStore as unknown as AbstractPrivateKeyStore);
+    this.musapKeyStore = keyStore;
+    this.sscdType = sscdType ? sscdType : 'TEE';
+    this.musapKeyStore.enableSscd(this.sscdType)
+  }
+
+  async listKeys(): Promise<ManagedKeyInfo[]> {
+    try {
+      const keysJson: MusapKey[] = (await this.musapKeyStore.listKeys()) as MusapKey[];
+      return keysJson.map((key) => this.asMusapKeyInfo(key));
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  async createKey(args: { type: TKeyType; sscdType?: SscdType }): Promise<ManagedKeyInfo> {
+    const keyAlgorithm = this.mapKeyTypeToAlgorithmType(args.type);
+
+    const keyGenReq: KeyGenReq = {
+      keyAlgorithm: keyAlgorithm,
+      did: '',
```

Comment:
What's up with this?

Reply:
this proprty is coming from the musap and is not present in the ios side. just filling it up here with an empty string. @sanderPostma will remove it from the musap-types and then I can remove it from here

Reply:
Okay but then create a comment about it

Reply:
fixed it in #40